Add an example for creating an automation to mark zombie flow runs as crashed (#16425)
PrefectHQ/prefect: docs/v3/automate/events/automations-triggers.mdx

- `user_id_1` creates and then completes an order, triggering a run of our deployment.
- `user_id_2` creates an order, but no completed event is emitted so no deployment is triggered.

### Detect and respond to zombie flows

If the infrastructure a flow is running on suddenly fails (e.g. the machine crashes or a container is evicted), 
Prefect's orchestration engine will be unable to report state changes and the flow run will get stuck in the running state.

Fortunately, flow runs triggered via deployment can emit heartbeats as they are running, and a Prefect automation can update 

> a flow run's state to crashed if the server stops receiving heartbeats for a flow run.

Reviewer: ```suggestion
a flow run's state to crashed if the server stops receiving heartbeats for that flow run.
```

<Note>
**Enable flow run heartbeat events**

> You will need to ensure you're running Prefect version 3.1.8 or greater and set `PREFECT_RUNNER_HEARTBEAT_FREQUENCY` to emit flow run heartbeat events.

Reviewer: worth including "set to an integer / to an integer greater than 30"?

- `user_id_1` creates and then completes an order, triggering a run of our deployment.
- `user_id_2` creates an order, but no completed event is emitted so no deployment is triggered.

### Detect and respond to zombie flows

> If the infrastructure a flow is running on suddenly fails (e.g. the machine crashes or a container is evicted), 

Reviewer 2: ```suggestion
If the infrastructure a flow is running on suddenly fails (for example, if the machine crashes or a container is evicted), 
```

Prefect's orchestration engine will be unable to report state changes and the flow run will get stuck in the running state.

Fortunately, flow runs triggered via deployment can emit heartbeats as they are running, and a Prefect automation can update 
a flow run's state to crashed if the server stops receiving heartbeats for that flow run.

<Note>
**Enable flow run heartbeat events**

You will need to ensure you're running Prefect version 3.1.8 or greater and set `PREFECT_RUNNER_HEARTBEAT_FREQUENCY`
to an integer greater than 30 to emit flow run heartbeat events.
</Note>

To create an automation that marks zombie flow runs as crashed, run this script:
```python
from datetime import timedelta

from prefect.automations import Automation
from prefect.client.schemas.objects import StateType
from prefect.events.actions import ChangeFlowRunState
from prefect.events.schemas.automations import EventTrigger, Posture
from prefect.events.schemas.events import ResourceSpecification

> 

Reviewer 2: ```suggestion

```

my_automation = Automation(
    name="Crash zombie flows",
    trigger=EventTrigger(
        after={"prefect.flow-run.heartbeat"},
        expect={
            "prefect.flow-run.heartbeat",
            "prefect.flow-run.Completed",
            "prefect.flow-run.Failed",
            "prefect.flow-run.Cancelled",
            "prefect.flow-run.Crashed",
        },
        match=ResourceSpecification({"prefect.resource.id": ["prefect.flow-run.*"]}),
        for_each={"prefect.resource.id"},
        posture=Posture.Proactive,
        threshold=1,
        within=timedelta(seconds=90),
    ),
    actions=[
        ChangeFlowRunState(
            state=StateType.CRASHED,
            message="Flow run marked as crashed due to missing heartbeats.",
        )
    ],
)

> my_automation.create()

Reviewer 2: ```suggestion
if __name__=="__main__":
    my_automation.create()
```
